OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -804,9 +804,8 @@ Activity LocalMove(Actor self, WPos fromPos, WPos toPos, CPos cell)
 
 		CPos? ClosestGroundCell()
 		{
-			var above = new CPos(TopLeft.X, TopLeft.Y);
-			if (CanEnterCell(above))
-				return above;
+			if (CanEnterCell(TopLeft))
+				return TopLeft;
```

Comment:
I didn't bother profiling it, so I'm not sure if keeping the `var above` but just populating it with `TopLeft` directly might be even better.

Reply:
I expect this will break things: the reason for the `new CPos(TopLeft.X, TopLeft.Y)` was to reset any custom `TopLeft.Layer` back to the default ground cell!